src/system/BlocksModule/Listener/BlockPositionInstallerListener.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Zikula package.
+ *
+ * Copyright Zikula - https://ziku.la/
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Zikula\BlocksModule\Listener;
+
+use Doctrine\Persistence\ManagerRegistry;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Zikula\BlocksModule\Entity\BlockPositionEntity;
+use Zikula\BlocksModule\Entity\RepositoryInterface\BlockPositionRepositoryInterface;
+use Zikula\ExtensionsModule\AbstractTheme;
+use Zikula\ExtensionsModule\Event\ExtensionPostInstallEvent;
+use Zikula\ExtensionsModule\Event\ExtensionStateEvent;
+
+/**
+ * When a new theme is installed, add non-existing block positions.
+ */
+class BlockPositionInstallerListener implements EventSubscriberInterface
+{
+    /**
+     * @var BlockPositionRepositoryInterface
+     */
+    private $blockPositionRepository;
+
+    /**
+     * @var ManagerRegistry
+     */
+    private $managerRegistry;
+
+    public function __construct(
+        BlockPositionRepositoryInterface $blockPositionRepository,
+        ManagerRegistry $managerRegistry
+    ) {
+        $this->blockPositionRepository = $blockPositionRepository;
+        $this->managerRegistry = $managerRegistry;
+    }
+
+    public static function getSubscribedEvents()
+    {
+        return [
+            ExtensionPostInstallEvent::class => [['createBlockPositions']],
```

Comment:
Is this notation (using two nested arrays) correct?

Reply:
yes - https://symfony.com/doc/current/event_dispatcher.html#creating-an-event-subscriber